src/main/org/pantsbuild/jarjar/MainProcessor.java

```diff
@@ -28,7 +29,8 @@
     private final KeepProcessor kp;
     private final Map<String, String> renames = new HashMap<String, String>();
     
-    public MainProcessor(List<PatternElement> patterns, boolean verbose, boolean skipManifest) {
```

Comment:
As mentioned above, since this constructor may already called from an external tool, let's leave it in place to preserve binary compatibility  and just have it delegate to the new constructor you've created below, defaulting `misplacedClassStrategy` to `SkipMisplacedClassStrategy(true)`.